src/main/java/me/ajaja/module/tag/adapter/out/persistence/CreateTagAdapter.java

```diff
@@ -0,0 +1,52 @@
+package me.ajaja.module.tag.adapter.out.persistence;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.RequiredArgsConstructor;
+import me.ajaja.module.tag.application.port.in.CreateTagUseCase;
+import me.ajaja.module.tag.application.port.out.CreateTagPort;
+import me.ajaja.module.tag.domain.FootprintTag;
+import me.ajaja.module.tag.domain.PlanTag;
+import me.ajaja.module.tag.domain.Tag;
+
+@Repository
+@Transactional
+@RequiredArgsConstructor
+public class CreateTagAdapter implements CreateTagPort {
+	private final TagRepository tagRepository;
+	private final FootprintTagRepository footprintTagRepository;
+	private final PlanTagRepository planTagRepository;
+
+	@Override
+	public void create(CreateTagUseCase.Type type, Long targetId, List<String> tagNames) {
+		Set<String> tagNameSet = new LinkedHashSet<>(tagNames);
+		List<Tag> savedTags = tagNameSet.stream().map(this::getOrCreateTagIfNotExists).toList();
+
+		switch (type) {
+			case FOOTPRINT -> saveFootprintTag(targetId, savedTags);
+			case PLAN -> savePlanTag(targetId, savedTags);
+		}
+	}
+
+	private Tag getOrCreateTagIfNotExists(String name) {
```

Comment:
뭔가 네이밍이 마음에 드는게 없네요? 조금 고민해봐도 좋을듯??
```suggestion
	private Tag getOrCreateIfNotExists(String name) {
```

Reply:
수정했습니다.

주변 코드도 많이보면서 신경을 더 써보겠습니다. 
위의 네이밍은 목적어를 붙이지 않으면 생성하는게 태그인지, 중간테이블인지 의도가 명확하지 않을 수 있어서 고민하다가 목적어를 붙였어요.

항상 의도를 얼마정도로 자세히 말해야할까가 가장 큰 고민인데 제 선택이 객관적으로 괜찮지는 않은 것 같습니다.

충분히 인지하고 있고 개선해보겠습니다.